htdocs/core/class/html.form.class.php

```diff
@@ -3980,7 +3980,17 @@ public function select_produits_fournisseurs_list($socid, $selected = '', $htmln
 		if (getDolGlobalInt('PRODUCT_USE_UNITS')) {
 			$sql .= " LEFT JOIN " . $this->db->prefix() . "c_units u ON u.rowid = p.fk_unit";
 		}
+		// we want to handle element sharing
+		if (isModEnabled('multicompany') && getDolGlobalInt('MULTICOMPANY_PRODUCT_SHARE_ALL_BY_DEFAULT')) {
+			$sql .= ' LEFT JOIN '.$this->db->prefix().'entity_element_sharing AS ees ON ees.fk_element = p.rowid AND ees.entity IN ('.getEntity("product").') AND ees.element = "product"';
```

Comment:
This tble is not a table of dolibarr, so thisnpaetbof code should be added by the hook of the module thatbmanage the table.